mobile/src/test/java/org/openhab/habdroid/model/OpenHAB2WidgetTest.java

```diff
@@ -1,67 +1,404 @@
+        sut1 = OpenHAB2Widget.createOpenHABWidgetFromJson(rootWidget, createJSONObject(1), "PNG");
+        sut2 = OpenHAB2Widget.createOpenHABWidgetFromJson(rootWidget, createJSONObject(2), "SVG");
+        sut3 = OpenHAB2Widget.createOpenHABWidgetFromJson(rootWidget, createJSONObject(3), "SVG");
+        assertTrue(sut1 instanceof OpenHAB2Widget);
+        assertTrue(sut2 instanceof OpenHAB2Widget);
     }
 
     @Test
-    public void testCreateOpenHABWidgetFromJson_createsOpenHAB2Widget() throws Exception {
-        OpenHABWidget sut = OpenHAB2Widget.createOpenHABWidgetFromJson(rootWidget, createJSONObject(), "PNG");
-        assertTrue(sut instanceof OpenHAB2Widget);
-    }
-
-    private JSONObject createJSONObject() throws Exception {
-
-        String json = "{\n" +
-                "    \"widgetId\": \"demo\",\n" +
-                "    \"type\": \"Group\",\n" +
-                "    \"label\": \"Group1\",\n" +
-                "    \"icon\": \"groupicon\",\n" +
-                "    \"url\": \"http://localhost/url\",\n" +
-                "    \"minValue\": \"0.0\",\n" +
-                "    \"maxValue\": \"10.0\",\n" +
-                "    \"step\": \"1\",\n" +
-                "    \"refresh\": \"10\",\n" +
-                "    \"period\": \"D\",\n" +
-                "    \"service\": \"D\",\n" +
-                "    \"height\": \"10\",\n" +
-                "    \"iconcolor\": \"white\",\n" +
-                "    \"labelcolor\": \"white\",\n" +
-                "    \"valuecolor\": \"white\",\n" +
-                "    \"mappings\": [{\n" +
-                "      \"command\": \"ON\",\n" +
-                "      \"label\": \"On\"\n" +
-                "    }],\n" +
-                "    \"item\": {\n" +
-                "      \"type\": \"GroupItem\",\n" +
-                "      \"name\": \"group1\",\n" +
-                "      \"state\": \"OFF\",\n" +
-                "      \"link\": \"http://localhost/rest/items/group1\"\n" +
-                "    },\n" +
-                "    \"linkedPage\": {\n" +
-                "      \"id\": \"0001\",\n" +
-                "      \"title\": \"LinkedPage\",\n" +
-                "      \"icon\": \"linkedpageicon\",\n" +
-                "      \"link\": \"http://localhost/rest/sitemaps/demo/0001\",\n" +
-                "      \"leaf\": \"false\"\n" +
-                "    },\n" +
-                "    \"widgets\": [{ \"widgetId\": \"demo11\" }]\n" +
-                "  }";
+    public void testGetIconPath() throws Exception {
+        assertEquals("icon/groupicon?state=OFF&format=PNG", sut1.getIconPath());
+        assertEquals("icon/groupicon?state=ON&format=SVG", sut2.getIconPath());
+        assertEquals("icon/slider?state=81&format=SVG", sut3.getChildren().get(0).getIconPath());
+
+        // Rollersutter icon should always be 0 to 100, not ON/OFF
+        //assertEquals("icon/rollershutter?state=0&format=SVG", sut3.getChildren().get(1).getIconPath());
```

Comment:
why is this line commented?

Reply:
Because this is currently broken and would lead to a failing travis build. When #542 I can rebase this PR and uncomment that line.

Reply:
Ah, ok, we should not forget to update one of the PR's to change this :)